apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -874,5 +886,28 @@ private void EnlistTransaction(Transaction transaction, TimestampBound timestamp
 
         /// <inheritdoc />
         protected override DbProviderFactory DbProviderFactory => SpannerProviderFactory.Instance;
+
+        /// <summary>
+        /// Get Resource based routing flag from environment variable.
+        /// </summary>
+        /// <returns>resource based routing flage</returns>
+        private static bool GetResourceBasedRoutingFlag()
+        {
+            string resourceBaseRoutingFlagValue =
+                Environment.GetEnvironmentVariable("GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING")?.ToLower();
+            return !string.IsNullOrEmpty(resourceBaseRoutingFlagValue) &&
+                resourceBaseRoutingFlagValue.Equals("true");
```

Comment:
nit: if you do `"true".Equals(....)` you can remove the check for null or empty.

Reply:
Or just `resourceBaseRoutingFlagValue == "true"`. This isn't Java ;)

Reply:
Or better yet:

```csharp
string resourceBaseRoutingFlagValue = Environment.GetEnvironmentVariable("GOOGLE_CLOUD_ENABLE_RESOURCE_BASED_ROUTING");
return StringComparer.InvariantCultureIgnoreCase.Equals(resourceBaseRoutingFlagValue, "true");
```

That way you avoid the culture-sensitive `ToLower()` call.